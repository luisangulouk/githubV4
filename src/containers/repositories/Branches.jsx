/* eslint-disable no-console */
import React from 'react';
import Style from './repositories.scss';

const Branches = ({
  refs
}) => {
  return (
    <ul className={Style.repoList}>
      {refs.nodes.map((node) => {
        console.log(node);
        return (
          <li key={node.id} className={Style.branchProfile}>
            <span className={Style.branchName}>{node.name}</span>
          </li>
        );
      }
      )}
    </ul>
  );
};

export default Branches;
