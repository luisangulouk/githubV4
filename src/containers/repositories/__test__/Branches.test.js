import React from 'react';
import renderer from 'react-test-renderer';

import Branches from '../Branches';

describe('Branches Component', () => {

  const mockRefs = {
    nodes: [{
      id: 'MDM6UmVmNjc1MjI1Nzc6cmVmcy9oZWFkcy9tYXN0ZXI=',
      name: 'master',
      __typename: 'Ref'
    }],
    __typename: 'RefConnection'
  };
  it('should match snapshot', () => {

    const component = renderer.create(<Branches refs={mockRefs} repo={'XamarinSQL'} login={'icebeam7'}/>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
