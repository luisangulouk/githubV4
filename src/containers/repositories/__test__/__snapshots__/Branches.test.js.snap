// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Branches Component should match snapshot 1`] = `
<ul
  className="repoList"
>
  <li
    className="branchProfile"
  >
    <div>
      <button
        className="btn btn-link branchNameLink"
        onClick={[Function]}
        type="button"
      >
        master
      </button>
    </div>
  </li>
</ul>
`;
